test/extensions/network/dns_resolver/cares/dns_impl_test.cc

```diff
@@ -1034,25 +1034,56 @@ TEST_P(DnsImplTest, LocalLookup) {
   EXPECT_NE(nullptr, resolveWithNoRecordsExpectation("foo.bar.baz", DnsLookupFamily::V4Only));
   dispatcher_->run(Event::Dispatcher::RunType::Block);
 
+  // For IPv4, expect exactly 127.0.0.1
   if (GetParam() == Address::IpVersion::v4) {
     EXPECT_EQ(nullptr, resolveWithExpectations("localhost", DnsLookupFamily::V4Only,
                                                DnsResolver::ResolutionStatus::Completed,
                                                {"127.0.0.1"}, {"::1"}, absl::nullopt));
   }
 
+  // For IPv6, verify that ::1 is present in results, but allow other addresses
```

Comment:
Can you clarify what other addresses are added for IPv6 localhost resolution, please?

Reply:
It's getting these two addresses: `{ "::1", "fd00:c098:190b:f9cf:7a8f:45d8:5ffb:2" }`

Reply:
@yanavlasov I tried implementing `ares_reinit()` and back-porting it in our private fork. I'm seeing major improvements in the number of timeouts and this version is a pre-requisite for implementing `ares_reinit()`. 

![image](https://github.com/user-attachments/assets/47618eb3-3e86-4a0b-a632-1435e498015d)

Whenever you have some time, could you please take another look? I'm happy to resolve any comments you may have.

Reply:
This change worries me a bit, since the localhost lookup returns the ULA address. I also do not see this failure on my dev setup. Was this failing just on your workstation or was a CI failure?